gennav/utils/custom_exceptions.py

```diff
@@ -0,0 +1,99 @@
+class Error(Exception):
+    """Base class for exceptions in this module."""
+    pass
+
+
+class Key_Error(Error):
+    """Exception raised when key is not found in the dictionary containing all parameters.
+
+    Args:
+        key (str): key to be searched in the dict.
+        kwargs (dict): dictionary containing all input parameters
+    """
+
+    def __init__(self, key, kwargs):
+        super().__init__()
+        self.key = key
+        self.dict = kwargs
+
+    def __str__(self):
+        try:
+            return "%s key not found in the dictionary containing all parameters (dict_keys[%s])" % (self.key, list(self.dict.keys()))
+        except SyntaxError:
+            return f"{self.key} key not found in the dictionary containing all parameters ({self.dict.keys()})."
+
+
+class StartStateinObs(Error):
+    """Exception raised when starting state is inside an obstacle.
+
+    Args:
+        start (gennav.utils.RobotState): Start state
+    """
+
+    def __init__(self, start, message="Start state is in obstacle."):
+        self.start = start
+        self.message = message
+        super(StartStateinObs, self).__init__()
+
+    def __str__(self):
+        try:
+            return "%s -> %s" % (self.start, self.message)
+        except SyntaxError:
+            return f"{self.start} -> {self.message}"
+
+
+class GoalStateinObs(Error):
```

Comment:
Similar name change here to `InvalidGoalState`

Reply:
Yeah.